packages/@aws-cdk/pipelines/lib/synths/simple-synth-action.ts

```diff
@@ -0,0 +1,353 @@
+
+  /**
+   * Name of the CodeBuild project
+   *
+   * @default - Automatically generated
+   */
+  readonly projectName?: string;
+
+  /**
+   * Build environment to use for CodeBuild job
+   *
+   * @default BuildEnvironment.LinuxBuildImage.STANDARD_1_0
+   */
+  readonly environment?: codebuild.BuildEnvironment;
+
+  /**
+   * Directory inside the source where package.json and cdk.json are located
+   *
+   * @default - Repository root
+   */
+  readonly subdirectory?: string;
+
+  /**
+   * Produce additional output artifacts after the build based on the given directories
+   *
+   * Can be used to produce additional artifacts during the build step,
+   * separate from the cloud assembly, which can be used further on in the
+   * pipeline.
+   *
+   * Directories are evaluated with respect to `subdirectory`.
+   *
+   * @default - No additional artifacts generated
+   */
+  readonly additionalArtifacts?: AdditionalArtifact[];
+}
+
+/**
+ * Construction props for SimpleSynthAction
+ */
+export interface SimpleSynthActionProps extends SimpleSynthOptions {
+  /**
+   * The synth command
+   */
+  readonly synthCommand: string;
+
+  /**
+   * The install command
+   *
+   * @default - No install required
+   */
+  readonly installCommand?: string;
+
+  /**
+   * The build command
+   *
+   * By default, we assume NPM projects are either written in JavaScript or are
+   * using `ts-node`, so don't need a build command.
+   *
+   * Otherwise, put the build command here, for example `npm run build`.
+   *
+   * @default - No build required
+   */
+  readonly buildCommand?: string;
+}
+
+/**
+ * Specification of an additional artifact to generate
+ */
+export interface AdditionalArtifact {
+  /**
+   * Directory to be packaged
+   */
+  readonly directory: string;
+
+  /**
+   * Artifact to represent the build directory in the pipeline
+   */
+  readonly artifact: codepipeline.Artifact;
+}
+
+/**
+ * A standard synth with a generated buildspec
+ */
+export class SimpleSynthAction implements codepipeline.IAction {
+
+  /**
+   * Create a standard NPM synth action
+   *
+   * Uses `npm ci` to install dependencies and `npx cdk synth` to synthesize.
+   *
+   * If you need a build step, add `buildCommand: 'npm run build'`.
+   */
+  public static standardNpmSynth(options: StandardNpmSynthOptions) {
+    return new SimpleSynthAction({
+      ...options,
+      installCommand: options.installCommand ?? 'npm ci',
+      synthCommand: options.synthCommand ?? 'npx cdk synth',
+    });
+  }
+
+  /**
+   * Create a standard Yarn synth action
+   *
+   * Uses `yarn install --frozen-lockfile` to install dependencies and `npx cdk synth` to synthesize.
+   *
+   * If you need a build step, add `buildCommand: 'yarn build'`.
+   */
+  public static standardYarnSynth(options: StandardYarnSynthOptions) {
+    return new SimpleSynthAction({
+      ...options,
+      installCommand: options.synthCommand ?? 'yarn install --frozen-lockfile',
```

Comment:
`installCommand` command is not overridable, should be as follows... as this is passing `options.synthCommand` to `installCommand`

`installCommand: options.installCommand ?? 'yarn install --frozen-lockfile',`

Reply:
@JeremyJonas Can you please raise an issue about this? Closed PRs are not ideal for tracking :-)